vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SearchManager.java

```diff
@@ -0,0 +1,267 @@
+        /*
+         * spring will just return an empty dataset if a page that is greater than the
+         * total number of pages is returned. So, the page given (except when it's < 1,
+         * then set it to 1) and then checking if the total page count is smaller than
+         * the given one (if so, make another request). In most cases the page number
+         * will be within the range of pages, so we would need only one db call. Only in
+         * cases where the page number is wrong, would a second one be needed.if
+         * page>total pages,last page is returned.
+         */
+        if (page > totalSlidePage) {
+            totalSlidePage = totalSlidePage == 0 ? 1 : totalSlidePage;
+            requestedPageForSlide = PageRequest.of(totalSlidePage - 1, pageSize);
+            slidePage = slideManager.findByNameOrDescription(requestedPageForSlide, searchTerm);
+        }
+        return slidePage;
+    }
+
+    /**
+     * Method to return the requested slides whose text blocks contains the search
+     * string
+     * 
+     * @param searchTerm string which has been searched
+     * @param page.      The page number starts from 1.
+     * @return list of slides whose text blocks contains the search string in the
+     *         requested page.
+     */
+    @Override
+    public Page<ISlide> searchInSlideTexts(String searchTerm, int page) {
+        /* if page<1, 1st page is returned */
+        if (page < 1) {
+            page = 1;
+        }
+        Pageable requestedPageForSlideText = PageRequest.of(page - 1, pageSize);
+        Page<ISlide> slidetextPage = textContentBlockRepo.findWithNameOrDescription(requestedPageForSlideText,
+                searchTerm);
+        int totalSlideTextPage = slidetextPage.getTotalPages();
+        /*
+         * spring will just return an empty dataset if a page that is greater than the
+         * total number of pages is returned. So, the page given (except when it's < 1,
+         * then set it to 1) and then checking if the total page count is smaller than
+         * the given one (if so, make another request). In most cases the page number
+         * will be within the range of pages, so we would need only one db call. Only in
+         * cases where the page number is wrong, would a second one be needed.if
+         * page>total pages,last page is returned.
+         */
+        if (page > totalSlideTextPage) {
+            totalSlideTextPage = totalSlideTextPage == 0 ? 1 : totalSlideTextPage;
+            requestedPageForSlideText = PageRequest.of(totalSlideTextPage - 1, pageSize);
+            slidetextPage = textContentBlockRepo.findWithNameOrDescription(requestedPageForSlideText, searchTerm);
+        }
+        return slidetextPage;
+    }
+    
+    protected abstract  Page<ISpace> spaceSearch(Pageable requestedPageForSpace ,  String searchTerm);
+
+    
+    @Override
+    public StaffSearchModuleResults getStaffSearchModuleResults(List<IModule> moduleList) {
```

Comment:
staff search related methods should be in the `StaffSearchManager`

Reply:
for all the `get...Results` methods, what are they for? It looks like one has to first search and then pass the search results to this method? Why? Unless there is a reason this should be two different methods called from another class, one call should be enough.

Reply:
the pagination.. methods which give the search results work differently for public search and staff search. hence both these are not merged into a single call.

Reply:
but why is this method in the searchmanager if it's a staff search specific method?

Reply:
I have renamed all the staff search methods and classes which are common to both public and staff